docs/source/cpp/compute.rst

```diff
@@ -563,30 +563,32 @@ representation based on the rounding criterion.
 +-------------------+------------+-------------+-------------------------+----------------------------------+--------+
 | floor             | Unary      | Numeric     | Float32/Float64/Decimal |                                  |        |
 +-------------------+------------+-------------+-------------------------+----------------------------------+--------+
-| round             | Unary      | Numeric     | Float32/Float64/Decimal | :struct:`RoundOptions`           | (1)(2) |
+| round             | Unary      | Numeric     | Input Type              | :struct:`RoundOptions`           | (1)(2) |
```

Comment:
So I think this is technically a breaking change right?

Before, if we had something like:

```
x = pa.array([1, 2], pa.int32())
y = pc.round(x)
```

Then `y` would be a double array. Now, `y` will be an `int32` array.  I think this is correct and the old behavior was unintentional so I think it is an ok breaking change.  Still, we should make sure to mark the PR as a breaking change if my understanding is correct so that we document it as such in the release notes.

CC @jorisvandenbossche for second opinion.

Reply:
Right, it should be a breaking change. @jorisvandenbossche could you please confirm if it's acceptable?

Reply:
Yes, fully agreed with the summary of Weston above: this was unintentional behaviour (because of automatic casting for numeric types), and it's fine to correct this with a breaking change.